botuser.php

```diff
@@ -221,8 +222,14 @@
                                 $DB = createDBObject();
                                 checkUserInDB($DB, $event["user"]);
                                 $userinfo = getUserInfo($DB,$event["user"]);
+								$userID = getUserInfo($id);
+								$mentor = getMentors($users["users"]);
                                 if(floor($event["ts"]) - $userinfo["cooldown"] > $TIMEBUCKS_RATE_LIMIT) {
-                                    $timebucks = $userinfo["balance"] + $TIMEBUCKS_INCREMENT;
+									if($userID == $mentor){
```

Comment:
These two objects are not ever going to be the same. Under your current implementation, $userID is a [user object](https://api.slack.com/types/user) while $mentor is [a list of user ids](https://api.slack.com/methods/usergroups.users.list)